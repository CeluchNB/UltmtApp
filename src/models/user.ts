import { Realm } from '@realm/react'

export const DisplayUserSchema: Realm.ObjectSchema = {
    name: 'DisplayUser',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string',
        localGuest: 'bool?',
    },
}
