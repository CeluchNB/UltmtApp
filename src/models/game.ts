import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
