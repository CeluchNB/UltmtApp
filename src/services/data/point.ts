import * as Constants from '../../utils/constants'
import { GuestUser } from '../../types/user'
import Point from '../../types/point'
import { throwApiError } from '../../utils/service-utils'
import { withGameToken } from './game'
import {
    createPoint as networkCreatePoint,
    setPlayers as networkSetPlayers,
} from '../network/point'

/**
 * Method to create a point
 * @param pulling boolean - calling team pulling or not
 * @param pointNumber number of point in the game
 * @returns created point
 */
export const createPoint = async (
    pulling: boolean,
    pointNumber: number,
): Promise<Point> => {
    try {
        const response = await withGameToken(
            networkCreatePoint,
            pulling,
            pointNumber,
        )

        const { point } = response.data
        return point
    } catch (e) {
        return throwApiError(e, Constants.CREATE_POINT_ERROR)
    }
}

/**
 * Method to select the players of a specific point
 * @param pointId id of point
 * @param players array of players (length must equal game's players per point value)
 * @returns updated point
 */
export const setPlayers = async (
    pointId: string,
    players: GuestUser[],
): Promise<Point> => {
    try {
        const response = await withGameToken(
            networkSetPlayers,
            pointId,
            players,
        )
        const { point } = response.data
        return point
    } catch (e) {
        return throwApiError(e, Constants.SET_PLAYERS_ERROR)
    }
}
