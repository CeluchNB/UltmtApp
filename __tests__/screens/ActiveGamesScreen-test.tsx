import * as GameData from '../../src/services/data/game'
import * as PointData from '../../src/services/data/point'
import * as UserData from '../../src/services/data/user'
import { ActiveGamesProps } from '../../src/types/navigation'
import ActiveGamesScreen from '../../src/screens/ActiveGamesScreen'
import { NavigationContainer } from '@react-navigation/native'
import { Provider } from 'react-redux'
import React from 'react'
import store from '../../src/store/store'
import { Game, LocalGame } from '../../src/types/game'
import { fetchProfileData, game, point } from '../../fixtures/data'
import {
    fireEvent,
    render,
    screen,
    waitFor,
} from '@testing-library/react-native'

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper')

const mockedNavigate = jest.fn()
const setOptions = jest.fn()
jest.mock('@react-navigation/native', () => {
    const actualNav = jest.requireActual('@react-navigation/native')
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigate,
        }),
    }
})

const createGame = (overrides: Partial<Game>): LocalGame => {
    return {
        ...game,
        offline: false,
        ...overrides,
    }
}

const props: ActiveGamesProps = {
    navigation: {
        mockedNavigate,
        setOptions,
        addListener: () => {
            return () => {}
        },
    } as any,
    route: {} as any,
}

beforeEach(() => {
    jest.resetAllMocks()
    jest.spyOn(GameData, 'getActiveGames').mockReturnValue(
        Promise.resolve([
            createGame({
                creator: {
                    _id: fetchProfileData._id,
                    firstName: fetchProfileData.firstName,
                    lastName: fetchProfileData.lastName,
                    username: fetchProfileData.username,
                },
            }),
            createGame({
                creator: {
                    _id: 'otheruser',
                    firstName: fetchProfileData.firstName,
                    lastName: fetchProfileData.lastName,
                    username: fetchProfileData.username,
                },
            }),
        ]),
    )
    jest.spyOn(UserData, 'getUserId').mockReturnValue(
        Promise.resolve(fetchProfileData._id),
    )
})

describe('ActiveGamesScreen', () => {
    it('renders', async () => {
        const snapshot = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )

        await waitFor(async () => {
            expect(
                screen.getByText(
                    `${game.teamOne.name} vs. ${game.teamTwo.name}`,
                ),
            ).toBeTruthy()
        })

        expect(screen.getByText(`vs. ${game.teamTwo.name}`)).toBeTruthy()
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('navigates without a point', async () => {
        jest.spyOn(GameData, 'resurrectActiveGame').mockReturnValueOnce(
            Promise.resolve({
                ...game,
                tournament: undefined,
                startTime: '2022',
                offline: false,
            } as unknown as Game & { offline: boolean }),
        )
        jest.spyOn(PointData, 'getActivePointForGame').mockReturnValueOnce(
            Promise.resolve(undefined),
        )
        const { getByText } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )
        await waitFor(async () => {
            expect(
                getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
            ).toBeTruthy()
        })

        fireEvent.press(
            getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
        )

        await waitFor(async () => {
            expect(mockedNavigate).toHaveBeenCalledWith('LiveGame', {
                screen: 'FirstPoint',
            })
        })
    })

    it('navigating with team one and no actions', async () => {
        jest.spyOn(GameData, 'resurrectActiveGame').mockReturnValueOnce(
            Promise.resolve({
                ...game,
                creator: {
                    _id: fetchProfileData._id,
                    firstName: fetchProfileData.firstName,
                    lastName: fetchProfileData.lastName,
                    username: fetchProfileData.username,
                },
                tournament: undefined,
                startTime: '2022',
                offline: false,
            } as unknown as LocalGame),
        )
        jest.spyOn(PointData, 'getActivePointForGame').mockReturnValueOnce(
            Promise.resolve(point),
        )
        const { getByText } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )
        await waitFor(async () => {
            expect(
                getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
            ).toBeTruthy()
        })

        fireEvent.press(
            getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
        )

        await waitFor(async () => {
            expect(mockedNavigate).toHaveBeenCalledWith('LiveGame', {
                screen: 'SelectPlayers',
            })
        })
    })

    it('navigating with team two and no actions', async () => {
        jest.spyOn(GameData, 'resurrectActiveGame').mockReturnValueOnce(
            Promise.resolve({
                ...game,
                creator: {
                    _id: 'nonuserid',
                    firstName: fetchProfileData.firstName,
                    lastName: fetchProfileData.lastName,
                    username: fetchProfileData.username,
                },
                tournament: undefined,
                startTime: '2022',
                offline: false,
            } as unknown as LocalGame),
        )
        jest.spyOn(PointData, 'getActivePointForGame').mockReturnValueOnce(
            Promise.resolve(point),
        )
        const { getByText } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )
        await waitFor(async () => {
            expect(
                getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
            ).toBeTruthy()
        })

        fireEvent.press(
            getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
        )

        await waitFor(async () => {
            expect(mockedNavigate).toHaveBeenCalledWith('LiveGame', {
                screen: 'SelectPlayers',
            })
        })
    })

    it('with actions', async () => {
        jest.spyOn(GameData, 'resurrectActiveGame').mockReturnValueOnce(
            Promise.resolve({
                ...game,
                creator: {
                    _id: fetchProfileData._id,
                    firstName: fetchProfileData.firstName,
                    lastName: fetchProfileData.lastName,
                    username: fetchProfileData.username,
                },
                tournament: undefined,
                startTime: '2022',
                offline: false,
            } as unknown as LocalGame),
        )
        jest.spyOn(PointData, 'getActivePointForGame').mockReturnValueOnce(
            Promise.resolve({ ...point, teamOneActions: ['an action'] }),
        )
        const { getByText } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )
        await waitFor(async () => {
            expect(
                getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
            ).toBeTruthy()
        })

        fireEvent.press(
            getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
        )

        await waitFor(async () => {
            expect(mockedNavigate).toHaveBeenCalledWith('LiveGame', {
                screen: 'LivePointEdit',
            })
        })
    })

    it('deletes a game', async () => {
        const spy = jest
            .spyOn(GameData, 'deleteGame')
            .mockReturnValueOnce(Promise.resolve(undefined))
        const { getByText, getAllByTestId } = render(
            <Provider store={store}>
                <NavigationContainer>
                    <ActiveGamesScreen {...props} />
                </NavigationContainer>
            </Provider>,
        )

        await waitFor(async () => {
            expect(
                getByText(`${game.teamOne.name} vs. ${game.teamTwo.name}`),
            ).toBeTruthy()
        })

        const deleteBtn1 = getAllByTestId('delete-button')[0]
        fireEvent.press(deleteBtn1)

        const confirmBtn = getByText('confirm')
        fireEvent.press(confirmBtn)

        expect(spy).toHaveBeenCalledTimes(1)

        const deleteBtn2 = getAllByTestId('delete-button')[0]
        fireEvent.press(deleteBtn2)

        const cancelBtn = getByText('cancel')
        fireEvent.press(cancelBtn)

        expect(spy).toHaveBeenCalledTimes(1)
    })
})
