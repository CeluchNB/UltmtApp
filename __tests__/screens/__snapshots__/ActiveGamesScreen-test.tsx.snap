// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveGamesScreen renders 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <RCTScrollView
      data={
        [
          {
            "_id": "game1",
            "creator": {
              "_id": "testid",
              "firstName": "first",
              "lastName": "last",
              "username": "testuser",
            },
            "floaterTimeout": true,
            "halfScore": 8,
            "hardcapMins": 90,
            "offline": false,
            "playersPerPoint": 7,
            "points": [],
            "resolveCode": "111111",
            "scoreLimit": 15,
            "softcapMins": 75,
            "startTime": 2022-10-12T00:00:00.000Z,
            "teamOne": {
              "_id": "team1",
              "name": "Temper",
              "place": "Pittsburgh",
              "seasonEnd": "2022",
              "seasonStart": "2022",
              "teamname": "pghtemper",
            },
            "teamOneActive": true,
            "teamOnePlayers": [],
            "teamOneScore": 3,
            "teamOneStatus": "active",
            "teamTwo": {
              "name": "Sockeye",
            },
            "teamTwoActive": false,
            "teamTwoDefined": false,
            "teamTwoPlayers": [],
            "teamTwoScore": 0,
            "teamTwoStatus": "guest",
            "timeoutPerHalf": 1,
            "totalViews": 0,
            "tournament": {
              "_id": "tourney1",
              "endDate": "2022-10-23",
              "eventId": "nationals22",
              "name": "Club Nationals 2022",
              "startDate": "2022-10-20",
            },
          },
          {
            "_id": "game1",
            "creator": {
              "_id": "otheruser",
              "firstName": "first",
              "lastName": "last",
              "username": "testuser",
            },
            "floaterTimeout": true,
            "halfScore": 8,
            "hardcapMins": 90,
            "offline": false,
            "playersPerPoint": 7,
            "points": [],
            "resolveCode": "111111",
            "scoreLimit": 15,
            "softcapMins": 75,
            "startTime": 2022-10-12T00:00:00.000Z,
            "teamOne": {
              "_id": "team1",
              "name": "Temper",
              "place": "Pittsburgh",
              "seasonEnd": "2022",
              "seasonStart": "2022",
              "teamname": "pghtemper",
            },
            "teamOneActive": true,
            "teamOnePlayers": [],
            "teamOneScore": 3,
            "teamOneStatus": "active",
            "teamTwo": {
              "name": "Sockeye",
            },
            "teamTwoActive": false,
            "teamTwoDefined": false,
            "teamTwoPlayers": [],
            "teamTwoScore": 0,
            "teamTwoStatus": "guest",
            "timeoutPerHalf": 1,
            "totalViews": 0,
            "tournament": {
              "_id": "tourney1",
              "endDate": "2022-10-23",
              "eventId": "nationals22",
              "name": "Club Nationals 2022",
              "startDate": "2022-10-20",
            },
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={[]}
      style={
        {
          "marginTop": 10,
        }
      }
      viewabilityConfigCallbackPairs={[]}
    >
      <View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#CEC6B4",
                        "fontSize": 41.0958904109589,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    vs. Sockeye
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 20.54794520547945,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    3 - 0
                  </Text>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 6,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="delete-button-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="delete-button-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityTraits="button"
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="delete-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#FFB4AB",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅖
                    </Text>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 6,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="go-button-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="go-button-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityTraits="button"
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="go-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ffd600",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅂
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onFocusCapture={[Function]}
          onLayout={[Function]}
          style={null}
        >
          <View>
            <View
              style={
                {
                  "display": "flex",
                  "flexDirection": "row",
                  "marginBottom": 5,
                }
              }
            >
              <View
                style={
                  {
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      {
                        "color": "#CEC6B4",
                        "fontSize": 41.0958904109589,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    Temper
                     vs. 
                    Sockeye
                  </Text>
                  <Text
                    style={
                      {
                        "color": "#ffd600",
                        "fontSize": 20.54794520547945,
                        "fontWeight": "bold",
                      }
                    }
                  >
                    3 - 0
                  </Text>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 6,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="delete-button-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="delete-button-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityTraits="button"
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="delete-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#FFB4AB",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅖
                    </Text>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": "transparent",
                    "borderRadius": 20,
                    "height": 40,
                    "margin": 6,
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                    "width": 40,
                  }
                }
                testID="go-button-container-outer-layer"
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "flex": 1,
                      "justifyContent": "center",
                      "overflow": "hidden",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                    }
                  }
                  testID="go-button-container"
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": false,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityTraits="button"
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="go-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ffd600",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅂
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      testID="base-modal"
      transparent={true}
      visible={false}
    />
  </View>
</RCTSafeAreaView>
`;
